Adjust flink sql connector to Apache Flink SQL Gateway

grzegorz8 opened the pull request:
## What changes were proposed in this pull request?

This pull request updates existing Flink SQL connector and makes it compatible with Apache Flink SQL Gateway.


This is still a draft pull request, since I encounter an error when I apply my changes directly to master branch. Namely, when I open Hue in the browser, in the logs I see:

```
[12/Feb/2025 00:37:18 -0800] api2         ERROR    Error running <function get_config at 0xffff4f1d3a60>
Traceback (most recent call last):
  File "/usr/share/hue/desktop/core/src/desktop/api2.py", line 108, in decorator
    return func(*args, **kwargs)
  File "/usr/share/hue/desktop/core/src/desktop/api2.py", line 137, in get_config
    config['storage_browser']['restrict_file_extensions'] = RESTRICT_FILE_EXTENSIONS.get()
AttributeError: 'Config' object has no attribute 'get'
```
However, the PR works well if the changes are applied on tag `release-4.11.0`.


There are a few **known issues** that are worth mentioning:

* Hue does not have concept of "catalogs". Therefore, Table Explorer tab and Autocompleter works well as long as you are working within one catalog. In particular, there will be no autocomplete hints for the following query (`|` is cursor position).
  `SELECT column_name_prefix| FROM catalog_name.database_name.table_name`.


* The red cancel button is visible even if the query has finished (e.g. in batch mode or when the query finishes instantly such as create table, use catalog etc).
  `<a class="snippet-side-btn red" data-bind="click: cancel, visible: status() == 'running' || status() == 'starting' || status() == 'available'" title="${ _('Cancel operation') }">`
  I'm wondering if we can make the button disappear if, for example, there are no more records to be returned (Gateway returns 'EOS' status). 

## How was this patch tested?

 - manual tests

ranade1 reviewed (approved):
Looks good.

bjornalm reviewed (approved):
Very nice work!

bjornalm commented:
@grzegorz8 thanks for contributing, check to see if you can resolve the conflicts and failing CI checks

Harshg999 commented on desktop/core/src/desktop/api_public_urls.py:
Are these changes required?

grzegorz8 replied:
No, they are not. As I mentioned in the description, I have difficulties to run image built from master (I would appreciate any help). So I applied https://github.com/cloudera/hue/pull/3866 manually. Once I'll rebase with master, these changes will not be needed.

grzegorz8 replied:
removed

Harshg999 commented on tools/docker/hue/Dockerfile:
```diff
@@ -6,6 +6,8 @@ LABEL description="Hue SQL Assistant - gethue.com"
 ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update -y && apt-get install -y \
+  pkg-config \
+  build-essential \
```
Are these changes required?

grzegorz8 replied:
No. Sorry, I'll remove it.

grzegorz8 replied:
removed

Harshg999 commented on desktop/libs/notebook/src/notebook/templates/editor_components.mako:
```diff
@@ -1729,7 +1729,7 @@ ${ sqlSyntaxDropdown.sqlSyntaxDropdown() }
     </div>
     <!-- ko if: !isCanceling() -->
-    <a class="snippet-side-btn red" data-bind="click: cancel, visible: status() == 'running' || status() == 'starting'" title="${ _('Cancel operation') }">
+    <a class="snippet-side-btn red" data-bind="click: cancel, visible: status() == 'running' || status() == 'starting' || status() == 'available'" title="${ _('Cancel operation') }">
```
What is the scenario when cancel needs to be enabled for available status?

grzegorz8 replied:
In fact it should be:
```
(type() == 'flink' && status() == 'available')
```

I'll try to explain. Hue starts fetching results when status is "available". Since Flink is streaming framework, the query may run infinitely, unless the user cancel it or Flink SQL gateway returns "END OF STREAM" status. This is why Cancel button needs to be visible when status is available.

I mentioned this problem in PR description.

> The red cancel button is visible even if the query has finished (e.g. in batch mode or when the query finishes instantly such as create table, use catalog etc).
<a class="snippet-side-btn red" data-bind="click: cancel, visible: status() == 'running' || status() == 'starting' || status() == 'available'" title="${ _('Cancel operation') }">

> I'm wondering if we can make the button disappear if, for example, there are no more records to be returned (Gateway returns 'EOS' status).

Unfortunately, I found it too difficult to manage it in Hue.

Harshg999 replied:
Makes sense, status available check should only with Flink type for now.
@ramprasadagarwal @bjornalm - Can anyone do one last review for this file?

bjornalm replied:
I think this is safe, showing a cancel button.

Harshg999 reviewed (commented):
Awesome work @grzegorz8! Thanks for your contribution!
I've left some minor review comments.

Also it feels like some of the issues you are getting are because of potential merge conflicts or bad rebase. Can you try rebasing to latest Hue master again please?

grzegorz8 commented:
> Also it feels like some of the issues you are getting are because of potential merge conflicts or bad rebase. Can you try rebasing to latest Hue master again please?

When I removed `filebrowser` from `app_blacklist`, it started working.

I rebased with master and pushed the changes to the PR.
